x/gamm/keeper/grpc_query.go

```diff
@@ -89,14 +89,104 @@ func (k Keeper) Records(ctx context.Context, req *types.QueryRecordsRequest) (*t
 	}, nil
 }
 
-func (k Keeper) SpotPrice(ctx context.Context, request *types.QuerySpotPriceRequest) (*types.QuerySpotPriceResponse, error) {
-	panic("implement me")
+func (k Keeper) SpotPrice(ctx context.Context, req *types.QuerySpotPriceRequest) (*types.QuerySpotPriceResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	if req.TokenInDenom == "" {
+		return nil, status.Error(codes.InvalidArgument, "invalid denom")
+	}
+
+	if req.TokenOutDenom == "" {
+		return nil, status.Error(codes.InvalidArgument, "invalid denom")
```

Comment:
`req.TokenInDenom` and `req.TokenOutDenom` have response.
A. Merge these two if statements
B. differentiate the response like "invalid out denom"